src/infrastructure/acknowlegement.rs

```diff
@@ -0,0 +1,42 @@
+use crate::net::{ExternalAcks, LocalAckRecord};
+
+/// Type responsible for handling the acknowledgement of packets.
+pub struct AcknowledgementHandler {
+    waiting_packets: LocalAckRecord,
+    their_acks: ExternalAcks,
+    pub seq_num: u16,
```

Comment:
At some point, it might be worth aliasing u16 to like SequenceNumber. We use it in a lot of places, and it might be clearer than just u16.

Reply:
Yea, good one.